### DISCUSSION TITLE
Front matter - return empty array instead of null

### DISCUSSION THREAD

**Discussion by joshbruce** [thephpleague/commonmark#735] [Ideas]
https://github.com/8fold/commonmark-fluent-markdown/issues/4

**Comment by colinodell** [comment]
Symfony's YAML component will return `null` for all of these inputs:

```php
$inputs = [
    '',
    '    ',
    "---\n",
    'null',
    "---\nnull",
    '# This is a comment',
];
```

The YAML spec does seem to indicate that `null` is a good return type to use in those cases:

 - https://yaml.org/spec/1.2.2/#72-empty-nodes
 - https://yaml.org/spec/1.2.2/#10211-null

> Believe it's easier to accommodate an empty version of a consistent type than it is to account for a type that represents absolute nothing. (I'm admittedly biased though as I tend to avoid null as much as possible.)

I agree that `null` should be avoided as much as possible, but if the YAML spec says that nulls are totally valid then I am attempted to respect that.

Also, while frontmatter is _almost_ always comprised of key-value pairs that can be represented as an array, we don't currently enforce that - any type of valid YAML document, even if it only contains a single scalar value:

```php
$inputs = [
    'Hello World!', // string
    '1', // int
    '1.0', // float
    '""', // empty string
];
```

As long as this remains the case then I think `null` is probably an okay "empty value" to return.  However, if we were to enforce that YAML must always contain an array (and throw exceptions for non-empty, non-array-like frontmatter), then I would be okay returning an empty array.

Definitely open to additional thoughts here, though!

**Reply by joshbruce** [reply]
Fair. Thanks for the pertinent specification links! :)
